### PR TITLE
Subtask UI

### LINKED ISSUE #82: Human readable cron hover display - Issue 5
We would like the cron field to be a human readable field. This is a usability feature. 

Priority: Low

### LINKED ISSUE #140: Subtask UI
image

* [x] Break up into 2 tabs: Coatue UI/Original

The Coatue UI has just 2 buttons
* [x] Link to Cloudwatch Logs
* [ ] Run this and all downstream
* [ ] Run this and no downstream

### PR DESCRIPTION
**Pull request by skim-coatue**
resolves #140  and resolves #82 

Separates task modal into Coatue UI And Original UI. Coatue UI give the option to run DAG from the selected task and shows a button to the Cloudwatch logs if it exists.

you can (temporarily) test the changes:

cloudwatch link: http://10.0.45.50:8080/admin/airflow/graph?dag_id=noaa_monthly_datasync&execution_date=

run from here button: http://10.0.45.50:8080/admin/airflow/graph?dag_id=jumpshot_spark&execution_date=

### CODE REVIEW

File: airflow/www/templates/airflow/dag.html [hunk 1]
```diff
@@ -131,8 +131,28 @@ <h4 class="modal-title" id="myModalLabel">
...
+              <form method="POST">
+                <input name="csrf_token" type="hidden" value="{{ csrf_token() }}"/>
+                <input name="dag_id" type="hidden">
+                <input name="task_id" type="hidden">
+                <input name="execution_date" type="hidden">
+                <input name="origin" type="hidden" value="{{ request.base_url }}">
+                <button id="btn_run_from_here" type="button" class="btn btn-primary"
+                        title="Run this task and all downstream tasks"
+                        data-action="{{ url_for('airflow.run_from_here') }}">
+                  Run Task and All Downstream Tasks
```
Comment (pathikrit) [inline airflow/www/templates/airflow/dag.html]: This should be renamed to `Retry this task and run all downstream`

File: airflow/www/views.py [hunk 2]
```diff
@@ -2296,30 +2337,35 @@ def ordering(dag):
...
+        :return: Dict of key-value pairs from XCom
+        """
+        XCom = models.XCom
+        task_id = request.args.get("task_id")
+        dag_id = request.args.get("dag_id")
+        dttm = pendulum.parse(request.args.get("dttm"))
+        xcomlist = session.query(XCom).filter(
+            XCom.dag_id == dag_id, XCom.task_id == task_id,
+            XCom.execution_date == dttm).all()
+        payload = {}
```
Comment (pathikrit) [inline airflow/www/views.py]: you can do set comprehension here btw e.g.:

```suggestion
        payload = {xcom.key: xcom.value for xcom in xcomlist}
```

**Comment by skim-coatue** [comment]
Hm that means that I'd have to manually modify the run_id in the database and to add the "trig_for_override_run__" prefix to it. This may have potential side effects that I'd have to test out so I'll save it for a different PR

**Comment by skim-coatue** [comment]
Screen Shot 2019-12-02 at 5 18 40 PM

Also, the `Trigger DAG` button there should probably also take you to Dynamic Runs, right?

**Comment by pathikrit** [comment]
> Also, the Trigger DAG button there should probably also take you to Dynamic Runs, right?

Yes

**Comment by pathikrit** [comment]
> Also, the Trigger DAG button there should probably also take you to Dynamic Runs, right?

Yes

**Comment by skim-coatue** [comment]
Sure I will make that change across all the hover things we have now in a separate PR. I want to avoid adding too much to 1 PR. 

I'll refactor to not use `title` which lets the browser set the font/hover time, etc. See issue here: https://github.com/coatue/airflow/issues/171

Also issue for the other comments above: https://github.com/coatue/airflow/issues/170

**Pull request merged**